hintanimedittext/src/main/java/com/xiaochendev/lib/HintAnimEditText.java

```diff
@@ -0,0 +1,155 @@
+package com.xiaochendev.lib;
+
+import ohos.agp.animation.Animator;
+import ohos.agp.animation.AnimatorValue;
+import ohos.agp.components.AttrSet;
+import ohos.agp.components.Component;
+import ohos.agp.components.TextField;
+import ohos.agp.render.Canvas;
+import ohos.agp.render.Paint;
+import ohos.agp.utils.Color;
+import ohos.agp.utils.TextAlignment;
+import ohos.agp.utils.TextTool;
+import ohos.app.Context;
+
+/**
+ * Created by tanfujun on 8/18/16.
+ *
+ * <p>Be careful not to use the original hint of EditText when using this control. Use the in-control method instead.
+ */
+public class HintAnimEditText extends TextField implements Component.DrawTask {
+
+    CharSequence mAnimHintString = "";
+    Paint mHintPaint;
+    float mDefaultAlpha;
+    float mDefaultTextSize;
+    float mTextLength;
+
+    public HintAnimEditText(Context context) {
+        super(context);
+        init();
+    }
+
+    public HintAnimEditText(Context context, AttrSet attrs) {
+        super(context, attrs);
+        init();
+    }
+
+    public HintAnimEditText(Context context, AttrSet attrs, int defStyleAttr) {
+        super(context, attrs, "defStyleAttr");
+        init();
+    }
+
+    private void init() {
+        mHintPaint = new Paint();
+        mHintPaint.setAntiAlias(true);
+        mHintPaint.setColor(getHintColor());
+        mHintPaint.setAlpha((int) getAlpha());
+        mHintPaint.setColor(Color.WHITE);
```

Comment:
Base library sets the color to GRAY. Also hint texts ideally will be in gray color. Please retain the base library implementation

Reply:
Done